<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VibrationPage</class>
 <widget class="QWidget" name="VibrationPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>振动监测系统</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* 全局设置 */
QWidget#VibrationPage {
    background-color: #f0f2f5;
    font-family: &quot;Segoe UI&quot;, &quot;Microsoft YaHei&quot;, sans-serif;
    color: #333;
}

/* 顶部控制栏 */
QFrame#controlFrame {
    background-color: #ffffff;
    border-bottom: 1px solid #dcdfe6;
}

/* 按钮通用样式 */
QPushButton {
    border: none;
    border-radius: 4px;
    padding: 6px 16px;
    font-size: 14px;
    font-weight: bold;
    color: white;
    min-height: 24px;
}

/* 绿色按钮 (开始) */
QPushButton#btn_start {
    background-color: #67c23a;
}
QPushButton#btn_start:hover {
    background-color: #85ce61;
}
QPushButton#btn_start:pressed {
    background-color: #5daf34;
}

/* 黄色按钮 (暂停) */
QPushButton#btn_pause {
    background-color: #e6a23c;
}
QPushButton#btn_pause:hover {
    background-color: #ebb563;
}
QPushButton#btn_pause:pressed {
    background-color: #cf9236;
}

/* 红色按钮 (停止) */
QPushButton#btn_stop {
    background-color: #f56c6c;
}
QPushButton#btn_stop:hover {
    background-color: #f78989;
}
QPushButton#btn_stop:pressed {
    background-color: #dd6161;
}

/* 禁用状态 */
QPushButton:disabled {
    background-color: #f4f4f5;
    border: 1px solid #dcdfe6;
    color: #bcbec2;
}

/* 图表分组框 - 紧凑型卡片 */
QGroupBox {
    background-color: #ffffff;
    border: 1px solid #e4e7ed;
    border-radius: 4px;
    margin-top: 10px;
    padding-top: 10px;
    font-weight: bold;
    font-size: 13px;
    color: #606266;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top left;
    left: 10px;
    padding: 0 5px;
    background-color: transparent; 
}

/* 底部状态栏 */
QFrame#statusFrame {
    background-color: #2b3137;
    border-top: 1px solid #444;
}

/* 状态栏标签 */
QLabel {
    font-size: 12px;
}
</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="controlFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>60</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>60</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <layout class="QHBoxLayout" name="controlLayout">
      <property name="spacing">
       <number>15</number>
      </property>
      <property name="leftMargin">
       <number>20</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>20</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QLabel" name="label_title">
        <property name="styleSheet">
         <string>font-size: 18px; font-weight: bold; color: #303133;</string>
        </property>
        <property name="text">
         <string>振动波形实时监测</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_separator">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_start">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>36</height>
         </size>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="text">
         <string>开始采集</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_pause">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>36</height>
         </size>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="text">
         <string>暂停显示</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_stop">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>36</height>
         </size>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="text">
         <string>停止并重置</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="plotsWidget" native="true">
     <layout class="QVBoxLayout" name="plotsVerticalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox_ch1">
        <property name="font">
         <font>
          <pointsize>-1</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="title">
         <string>通道 1 (X-Axis) - 加速度</string>
        </property>
        <layout class="QVBoxLayout" name="ch1Layout">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>15</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <widget class="QCustomPlot" name="plot_ch1" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>200</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_ch2">
        <property name="title">
         <string>通道 2 (Y-Axis) - 加速度</string>
        </property>
        <layout class="QVBoxLayout" name="ch2Layout">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>15</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <widget class="QCustomPlot" name="plot_ch2" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>200</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_ch3">
        <property name="title">
         <string>通道 3 (Z-Axis) - 加速度</string>
        </property>
        <layout class="QVBoxLayout" name="ch3Layout">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>15</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <widget class="QCustomPlot" name="plot_ch3" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>200</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="statusFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>35</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>35</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <layout class="QHBoxLayout" name="statusLayout">
      <property name="spacing">
       <number>15</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label_status_icon">
        <property name="maximumSize">
         <size>
          <width>12</width>
          <height>12</height>
         </size>
        </property>
        <property name="styleSheet">
         <string>background-color: #67c23a; border-radius: 6px;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_status">
        <property name="styleSheet">
         <string>color: #eeeeee; font-weight: bold;</string>
        </property>
        <property name="text">
         <string>就绪</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="statusSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_statistics">
        <property name="styleSheet">
         <string>font-family: &quot;Consolas&quot;, &quot;Courier New&quot;, monospace; color: #409eff; font-size: 13px;</string>
        </property>
        <property name="text">
         <string>采样率: 0 Hz | 数据包: 0 | 丢包: 0%</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
